src/System.Text.Json/System/Text/Json/JsonParseObject.cs

```diff
@@ -182,6 +182,19 @@ public bool TryGetValue(string propertyName, out JsonObject value)
             }
         }
 
+        public int Length
+        {
+            get
+            {
+                var record = Record;
+                if (record.Type != JsonValueType.Array)
+                {
+                    throw new NullReferenceException();
```

Comment:
InvalidOperationException is more appropriate. Feel free to change the exception type. Tyvm :)